Upstream-Status: Inappropriate

There were many attempts to upstream the same kind of patch,
but nobody succeded to do so due to some consistency problems.

We, at igalia, have been keeping this patch in downstream at
github.com/igalia/chromium in the ozone-wayland-dev branch
since M65, and constantly rebasing it. In the future, we will
try to upstream it.

Signed-off-by: Maksim Sisov <msisov@igalia.com>
---
From b07c0310381dd1f83b7db5d9daf7b768df6f252b Mon Sep 17 00:00:00 2001
From: Damian Hobson-Garcia <dhobsong@igel.co.jp>
Date: Wed, 21 Mar 2018 13:18:17 +0200
Subject: [PATCH 08/42] Add mmap via libv4l to generic_v4l2_device

Issue #437
---
 media/gpu/v4l2/generic_v4l2_device.cc | 10 ++++++++++
 media/gpu/v4l2/v4l2.sig               |  2 ++
 2 files changed, 12 insertions(+)

diff --git a/media/gpu/v4l2/generic_v4l2_device.cc b/media/gpu/v4l2/generic_v4l2_device.cc
index 42d2cb2a6051..f07bdf2b30ad 100644
--- a/media/gpu/v4l2/generic_v4l2_device.cc
+++ b/media/gpu/v4l2/generic_v4l2_device.cc
@@ -100,10 +100,20 @@ void* GenericV4L2Device::Mmap(void* addr,
                               int flags,
                               unsigned int offset) {
   DCHECK(device_fd_.is_valid());
+#if BUILDFLAG(USE_LIBV4L2)
+  if (use_libv4l2_)
+    return v4l2_mmap(addr, len, prot, flags, device_fd_.get(), offset);
+#endif
   return mmap(addr, len, prot, flags, device_fd_.get(), offset);
 }
 
 void GenericV4L2Device::Munmap(void* addr, unsigned int len) {
+#if BUILDFLAG(USE_LIBV4L2)
+  if (use_libv4l2_) {
+    v4l2_munmap(addr, len);
+    return;
+  }
+#endif
   munmap(addr, len);
 }
 
diff --git a/media/gpu/v4l2/v4l2.sig b/media/gpu/v4l2/v4l2.sig
index 4269fb48d592..71b5b3787d56 100644
--- a/media/gpu/v4l2/v4l2.sig
+++ b/media/gpu/v4l2/v4l2.sig
@@ -8,3 +8,5 @@
 LIBV4L_PUBLIC int v4l2_close(int fd);
 LIBV4L_PUBLIC int v4l2_ioctl(int fd, unsigned long int request, ...);
 LIBV4L_PUBLIC int v4l2_fd_open(int fd, int v4l2_flags);
+LIBV4L_PUBLIC void *v4l2_mmap(void *start, size_t length, int prot, int flags, int fd, int64_t offset);
+LIBV4L_PUBLIC int v4l2_munmap(void *_start, size_t length);
-- 
2.17.1

